controllers/compassmanager_controller.go

```diff
@@ -428,3 +306,198 @@ func createCompassRuntimeLabels(kymaLabels map[string]string) map[string]interfa
+		kymaCR.Labels = make(map[string]string)
+	}
+	if kymaCR.Annotations == nil {
+		kymaCR.Annotations = make(map[string]string)
+	}
+
+	c.cache.kymaCR = &kymaCR
+
+	return c.cache.kymaCR, nil
+}
+
+// GetCompassMapping return nil, nil if it doesn't exist
+func (c *ClusterInterface) GetCompassMapping(name types.NamespacedName) (*v1beta1.CompassManagerMapping, error) {
+	if c.cache.compassMapping != nil && c.cache.compassMapping.Labels[LabelKymaName] == name.Name {
+		return c.cache.compassMapping, nil
+	}
+
+	mappingList := &v1beta1.CompassManagerMappingList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		LabelKymaName: name.Name,
+	})
+
+	err := c.kubectl.List(context.TODO(), mappingList, &client.ListOptions{
+		LabelSelector: labelSelector,
+		Namespace:     name.Namespace,
+	})
+
+	if err != nil || len(mappingList.Items) == 0 {
+		return nil, err
+	}
+
+	c.cache.compassMapping = &mappingList.Items[0]
+
+	if c.cache.compassMapping.Labels == nil {
+		c.cache.compassMapping.Labels = make(map[string]string)
+	}
+	if c.cache.compassMapping.Annotations == nil {
+		c.cache.compassMapping.Annotations = make(map[string]string)
+	}
+
+	return c.cache.compassMapping, nil
+}
+
+func (c *ClusterInterface) DeleteCompassMapping(name types.NamespacedName) error {
+	mapping, err := c.GetCompassMapping(name)
+	if err != nil {
+		return err
+	}
+	return c.kubectl.Delete(context.TODO(), mapping)
+}
+
+// GetKubeconfig return nil, nil if it doesn't exist
+func (c *ClusterInterface) GetKubeconfig(name types.NamespacedName) ([]byte, error) {
+	if c.cache.kubecfg != nil && c.cache.kubecfg.Labels[LabelKymaName] == name.Name {
+		return c.cache.kubecfg.Data[KubeconfigKey], nil
+	}
+
+	secretList := &corev1.SecretList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		LabelKymaName: name.Name,
+	})
+
+	err := c.kubectl.List(context.TODO(), secretList, &client.ListOptions{
+		LabelSelector: labelSelector,
+		Namespace:     name.Namespace, // TODO: Is this the same namespace?
+	})
+
+	if err != nil || len(secretList.Items) == 0 {
+		return nil, err
+	}
+
+	_, ok := secretList.Items[0].Data[KubeconfigKey]
+	if !ok {
+		return nil, nil
+	}
+
+	c.cache.kubecfg = &secretList.Items[0]
+
+	return c.cache.kubecfg.Data[KubeconfigKey], nil
+}
+
+func (c *ClusterInterface) UpsertCompassMapping(name types.NamespacedName, compassRuntimeID string) error {
+	kyma, err := c.GetKyma(name)
+	if err != nil {
+		return err
+	}
+
+	compassMapping := &v1beta1.CompassManagerMapping{}
+
+	compassMapping.Name = name.Name
+	compassMapping.Namespace = name.Namespace
+
+	labels := make(map[string]string)
+	labels[LabelKymaName] = kyma.Labels[LabelKymaName]
+	labels[LabelCompassID] = compassRuntimeID
+	labels[LabelGlobalAccountID] = kyma.Labels[LabelGlobalAccountID]
+	labels[LabelSubaccountID] = kyma.Labels[LabelSubaccountID]
+	labels[LabelManagedBy] = "compass-manager"
+
+	compassMapping.Labels = labels
+
+	// TODOs add retry for upsert logic
```

Comment:
Could you delete this comment? It is obsolete, we don't want another retry logic.